index.php

```diff
@@ -94,10 +94,34 @@
         ));
     }
 
+    global $CFG;
+    if (!empty($CFG->rememberusername)) {
+        $cookiename = 'REMEMBERUNAME';
```

Comment:
can we define a constant for 'REMEMBERUNAME' cookie, as we used this on multiple places in same file.
also if block can be written as 
define constant for 
const  COOKIE_REMEMBERUNAME = 'REMEMBERUNAME'; // define class const or use define function for it
if (!empty($CFG->rememberusername) && isset($_COOKIE[self::COOKIE_REMEMBERUNAME]) ) {
        global $frm;
        $frm->username =  rc4decrypt($_COOKIE[self::COOKIE_REMEMBERUNAME]);;
        $passwordform->set_data( [ 'username' => $frm->username, 'rememberme' => 1 ] );
    }

Reply:
Done, Please review.